From a614ca19bf8dce0122a4b15f7bf40c548a6968af Mon Sep 17 00:00:00 2001
From: Marek Belisko <marek.belisko@streamunlimited.com>
Date: Fri, 11 Apr 2014 13:55:01 +0200
Subject: [PATCH] stream800: Fix problems with compilation

Without that change we get:
include/asm/arch/hardware.h: Assembler messages:
include/asm/arch/hardware.h:64: Error: bad instruction `int clk_get(int clk)'

It's common to not include cpu and other headers to include file.

Signed-off-by: Marek Belisko <marek.belisko@streamunlimited.com>
---
 board/streamunlimited/stream800/adc.c                           | 3 +++
 board/streamunlimited/stream800/board-sue_demo_client_board.c   | 1 +
 board/streamunlimited/stream800/board-sue_factory_test_board.c  | 1 +
 board/streamunlimited/stream800/board-sue_first_carrier_board.c | 1 +
 include/configs/stream800.h                                     | 2 --
 5 files changed, 6 insertions(+), 2 deletions(-)

diff --git a/board/streamunlimited/stream800/adc.c b/board/streamunlimited/stream800/adc.c
index 941e91b..99fc405 100644
--- a/board/streamunlimited/stream800/adc.c
+++ b/board/streamunlimited/stream800/adc.c
@@ -20,6 +20,7 @@
 
 #include <common.h>
 #include <asm/io.h>
+#include <asm/arch/cpu.h>
 #include <asm/arch/clock.h>
 #include <asm/arch/hardware.h>
 
@@ -78,6 +79,8 @@
 #define REG_DMAREQ(X)			(0xec	+ ((X) * 0xc))
 #define REG_FIFODATA(X)			(0x100	+ ((X) * 0x100))
 
+//static const struct cm_wkuppll *cmwkup = (struct cm_wkuppll *)CM_WKUP;
+
 static unsigned long int adc_readl(unsigned long reg)
 {
 	return readl(TSC_ADC_SS_REG_BASE + reg);
diff --git a/board/streamunlimited/stream800/board-sue_demo_client_board.c b/board/streamunlimited/stream800/board-sue_demo_client_board.c
index ae8430c..9d5f363 100755
--- a/board/streamunlimited/stream800/board-sue_demo_client_board.c
+++ b/board/streamunlimited/stream800/board-sue_demo_client_board.c
@@ -22,6 +22,7 @@
 
 #include <common.h>
 #include <asm/io.h>
+#include <asm/arch/cpu.h>
 #include <asm/arch/clock.h>
 #include <asm/arch/hardware.h>
 #include <phy.h>
diff --git a/board/streamunlimited/stream800/board-sue_factory_test_board.c b/board/streamunlimited/stream800/board-sue_factory_test_board.c
index b7e2b3e..7438452 100644
--- a/board/streamunlimited/stream800/board-sue_factory_test_board.c
+++ b/board/streamunlimited/stream800/board-sue_factory_test_board.c
@@ -22,6 +22,7 @@
 
 #include <common.h>
 #include <asm/io.h>
+#include <asm/arch/cpu.h>
 #include <asm/arch/clock.h>
 #include <asm/arch/hardware.h>
 #include <phy.h>
diff --git a/board/streamunlimited/stream800/board-sue_first_carrier_board.c b/board/streamunlimited/stream800/board-sue_first_carrier_board.c
index 846bc40..f59a79b 100644
--- a/board/streamunlimited/stream800/board-sue_first_carrier_board.c
+++ b/board/streamunlimited/stream800/board-sue_first_carrier_board.c
@@ -22,6 +22,7 @@
 
 #include <common.h>
 #include <asm/io.h>
+#include <asm/arch/cpu.h>
 #include <asm/arch/clock.h>
 #include <asm/arch/hardware.h>
 #include <phy.h>
diff --git a/include/configs/stream800.h b/include/configs/stream800.h
index afec816..a8c6967 100644
--- a/include/configs/stream800.h
+++ b/include/configs/stream800.h
@@ -22,8 +22,6 @@
 #define CONFIG_NAND
 
 #include <configs/ti_am335x_common.h>
-#include <asm/arch/cpu.h>
-#include <asm/arch/hardware.h>
 
 #define CONFIG_DMA_COHERENT
 #define CONFIG_DMA_COHERENT_SIZE	(1 << 20)
-- 
2.7.4

